src/main/java/com/syncstream_dev/syncstream/model/user/User.java

```diff
@@ -0,0 +1,62 @@
+package com.syncstream_dev.syncstream.model.user;
+
+import java.time.LocalDateTime;
+
+
+public class User {
+    private UserInfo userInfo;
+    private String uid;     // User id
+    private String username;
+    private int messagesSent;
+    private int points;
+
+    // Register User Constructor
+    public User(String uid, String username, String token, String email, String password, String avatarUrl) {
+        // Todo: Database integration
+        this.userInfo = new UserInfo(token, email, password, LocalDateTime.now(), LocalDateTime.now(), avatarUrl);
+        this.uid = uid;
+        this.username = username;
+        this.messagesSent = 0;
+        this.points = 0;
+    }
+
+    // Login User Constructor
+    public User(String username, String password) {
+        throw new UnsupportedOperationException("Not implemented yet.");    // Todo: Requires database integration to load user info
+    }
+
+    // Getters
+    public String getUserToken() {
+        return userInfo.getToken();
+    }
+
+    public String getUid() {
+        return uid;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public int getMessagesSent() {
+        return messagesSent;
+    }
+
+    public int getPoints() {
+        return points;
+    }
+
+    // Overrides
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) return true;
+        if (obj == null || getClass() != obj.getClass()) return false;
```

Comment:
For future reference:
```java
public boolean equals(Object obj) {
     if (this == obj) return true; // If it's the same instance, they are equal
     if (obj instanceof User) { // Check if the object is of type User
          User user = (User) obj; // Cast the object to User type
          return Objects.equals(this.uid, user.uid); // Check if uid fields are equal
     }
     return false; // If the object is not of type User, they are not equal
}
```

Reply:
753ba79180124ce26cca6b1553c442febd749ad7 addresses this